scripts/convert_data.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python
+
+# Data conversion utility for ner-uk data set. Originally created for purposes of training Stanza model.
+# Later Stanza has merged some of this code inside.
+# Terminology:
+# IOB https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging) Some sources refer to this as BIO
+# BEIOS - extended IOB with E == end (for multi word tokens)
+
+
+import argparse
+import logging
+import os
+import glob
+import time
+from collections import namedtuple
+import re
+from typing import Tuple
+
+from tqdm import tqdm
+from random import choices, shuffle
+
+BsfInfo = namedtuple('BsfInfo', 'id, tag, start_idx, end_idx, token')
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+
+
+def format_token_as_beios(token: str, tag: str) -> list:
+    t_words = token.split()
+    res = []
+    if len(t_words) == 1:
+        res.append(token + ' S-' + tag)
+    else:
+        res.append(t_words[0] + ' B-' + tag)
+        for t_word in t_words[1: -1]:
+            res.append(t_word + ' I-' + tag)
+        res.append(t_words[-1] + ' E-' + tag)
+    return res
+
+
+def format_token_as_iob(token: str, tag: str) -> list:
+    t_words = token.split()
+    res = []
+    if len(t_words) == 1:
+        res.append(token + ' B-' + tag)
+    else:
+        res.append(t_words[0] + ' B-' + tag)
+        for t_word in t_words[1:]:
+            res.append(t_word + ' I-' + tag)
+    return res
+
+
+def convert_bsf(data: str, bsf_markup: str, converter: str = 'beios') -> str:
+    """
+    Convert data file with NER markup in Brat standoff Format to BEIOS or IOB format.
+
+    :param converter: iob or beios converter to use for document
+    :param data: tokenized data to be converted. Each token separated with a space
+    :param bsf_markup: Brat standoff Format markup
+    :return: data in BEIOS or IOB format https://en.wikipedia.org/wiki/Inside–outside–beginning_(tagging)
+    """
+
+    def join_simple_chunk(chunk: str) -> list:
+        if len(chunk.strip()) == 0:
+            return []
+        # keep the newlines, but discard the non-newline whitespace
+        tokens = re.split(r'(\n)|\s', chunk.strip())
+        # the re will return None for splits which were not caught in a group
+        tokens = [x for x in tokens if x is not None]
```

Comment:
You might use `filter(None, tokens)` (don't forget that it returns iterator, not the list)

Reply:
Dove deep into the topic - looks like a religious debate :)
https://stackoverflow.com/questions/3013449/list-comprehension-vs-lambda-filter
I Will keep it as is - as this is also the code moved from Stanza